File: mars_rover/js/src/rover.js
class Rover {
  constructor(x, y, orientation) {
    this.x = x
    this.y = y
    this.orientation = orientation
  }

  move(movs) {
    movs.forEach(m => {
      if (m === "f") {
        this.y = this.y + 1
      } else if (m === "b") {
        this.y = this.y - 1
      }
    })
  }
}

module.exports = Rover

File: mars_rover/ruby/lib/rover.rb
class Rover
  attr_reader :x, :y, :orientation

  def initialize(start_x, start_y, orientation)
      @x = start_x
      @y = start_y
      @orientation = orientation
  end

  def move(movs)
    movs.each do |m|
        if @orientation == :north
          if m == 'f'
              @y = @y + 1
          elsif m == 'b'
              @y = @y - 1
          end
        end
      end
    end
  end